cmd/sample/tag-created-vms/README.md

```diff
@@ -0,0 +1,132 @@
+# Tag Created VMs
+
+This sample shows how to do several things:
+1. Listen to VM Created Events,
+2. Bind vSphere credentials into our Service, and
+3. Combine the above to tag newly created VMs with a precreated label.
+
+## Setup
+
+This sample assumes that you have already:
+1. Installed Knative Serving/Eventing,
+2. Stood up `ko apply -f vcsim.yaml` to simulate a vSphere endpoint
+   depositing events on the default Broker,
+3. Created our tag with `ko create -f tags.yaml`.
+
+
+## Triggering on VM Creations
+
+To start our tour, let's first look at how we consume VM creation events (a subset of
+what our setup is putting on the Broker).
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: to-sample
+spec:
+  filter:
+    attributes:
+      type: com.vmware.vsphere.vmcreatedevent
+  subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: sample
+```
+
+This `Trigger` resource sets up a standing query on the default `Broker` for any Cloud Events
+with type `com.vmware.vsphere.vmcreatedevent`, and forwards all that it finds on to our sample
+`Service`.
+
+
+## Binding vSphere credentials into our Service.
+
+To allow our Service to call back into the vSphere API we bind credential data into
+our `Service` with:
+```yaml
+apiVersion: sources.knative.dev/v1alpha1
+kind: VSphereBinding
+metadata:
+  name: sample-binding
+spec:
+  # Our Service is going to receive the credential data.
+  subject:
+    apiVersion: serving.knative.dev/v1
+    kind: Service
+    name: sample
+
+  # What vSphere endpoint to connect to, and the credentials to use.
+  address: https://vcsim.default.svc.cluster.local
+  skipTLSVerify: true
+  secretRef:
+    name: vsphere-credentials
+```
+
+In our Go code we can use this to get an authenticated client by writing:
+```go
+
+import "github.com/mattmoor/vmware-sources/pkg/vsphere"
+
+...
+
+client, err := vsphere.NewREST(ctx)
+if err != nil {
+	log.Fatal(err.Error())
+}
+```
+
+This sets up a `rest.Client`, but we also expose `vsphere.New` for getting a `govmomi.Client`.
+
+
+## Tagging newly created VMs
+
+Now let's put it all together:
+```go
+package main
+
+import (
+	"context"
+	"log"
+
+	cloudevents "github.com/cloudevents/sdk-go/v1"
```

Comment:
v1 -> v2

Reply:
still v1

Reply:
I knew I'd regret inlining that file, that didn't take long.